vnext/local-cli/runWindows/utils/VSProjectUtils.ps1

```diff
@@ -0,0 +1,153 @@
+}
+
+# Loads the packages.config file at $PackagesConfigPath
+# If $CheckConfig, check the file exists first
+function Load-PackagesConfigXml {
+    param(
+        [string] $PackagesConfigPath,
+        [bool] $CheckConfig = $True
+    )
+
+    if ($CheckConfig) {
+        Check-PackagesConfigXml -PackagesConfigPath $PackagesConfigPath
+    }
+
+    Write-Host
+    Write-Host "Loading " -NoNewLine
+    Write-Host -ForegroundColor White $PackagesConfigPath -NoNewLine
+    Write-Host "..." -NoNewLine
+
+    [xml]$content = Get-Content -Path $PackagesConfigPath
+
+    Write-Host " done."
+
+    $content.packages.package | foreach {
+            Write-Host "Found package " -NoNewLine
+            Write-Host -ForegroundColor White $_.id -NoNewLine
+            Write-Host " version " -NoNewLine
+            Write-Host -ForegroundColor White $_.version -NoNewLine
+            Write-Host "."
+        }
+
+    return $content;
+}
+
+# Update package $PackageName to version $PackageVersion in the packages.config
+# file at $PackagesConfigPath
+# If $CheckConfig, check the file exists first
+function Update-PackagesConfigXml {
+    param(
+        [string] $PackagesConfigPath,
+        [string] $PackageName,
+        [string] $PackageVersion,
+        [bool] $CheckConfig = $True
+    )
+
+    if ($CheckConfig) {
+        Check-PackagesConfigXml -PackagesConfigPath $PackagesConfigPath
+    }
+
+    Write-Host
+    Write-Host "Updating package " -NoNewLine
+    Write-Host -ForegroundColor White $PackageName -NoNewLine
+    Write-Host " to version " -NoNewLine
+    Write-Host -ForegroundColor White $PackageVersion -NoNewLine
+    Write-Host "..."
+
+    & nuget.exe update "$PackagesConfigPath" -Id "$PackageName" -Version "$PackageVersion"
+}
+
+# Add package $PackageName version $PackageVersion to the packages.config
+# file at $PackagesConfigPath
+# If $CheckConfig, check the file exists first
+function Add-PackagesConfigXml {
+    param(
+        [string] $PackagesConfigPath,
+        [string] $PackageName,
+        [string] $PackageVersion,
+        [bool] $CheckConfig = $True
+    )
+
+    $packageXml = Load-PackagesConfigXml -PackagesConfigPath $PackagesConfigPath
+
+    $existingPackageNode = $packageXml.selectSingleNode("packages/package[@id=""$PackageName""]")
+
+    if ($existingPackageNode) {
+        Write-Host
+        Write-Host "Nothing to add, package " -NoNewLine
+        Write-Host -ForegroundColor White $PackageName -NoNewLine
+        Write-Host " is already present."
+    } else {
+        Write-Host
+        Write-Host "Adding package " -NoNewLine
+        Write-Host -ForegroundColor White $PackageName -NoNewLine
+        Write-Host "..." -NoNewLine
+
+        $packagesNode = $packageXml.selectSingleNode('packages')
+        $newPackageNode = $packagesNode.appendChild($packageXml.CreateElement('package'))
+        $newPackageNode.SetAttribute('id', $PackageName)
+        $newPackageNode.SetAttribute('version', '0')
```

Comment:
'0' is a bug?

Reply:
no. nuget does not support adding a new package to the config. nuget can only update an existing package to a different version.

if i set the version to the correct version, then nuget will think it's a no-op and do nothing. by setting it to 0, then calling nuget update with the correct version, the correct version will be installed and all of the project updates (adding targets, props, whatever) will be processed.

Reply:
thanks - this is worth describing in a comment :)